test/contracts/ContractInterface.h

```diff
@@ -67,8 +67,8 @@ class ContractInterface
 		bytes const& ret = call(_name + "(string)", u256(0x20), _arg.length(), _arg);
 		BOOST_REQUIRE(ret.size() == 0x20);
 		BOOST_CHECK(std::count(ret.begin(), ret.begin() + 12, 0) == 12);
-		// TODO: simplify this
-		return u160(u256(util::h256(ret)));
+		bytes const addr{ret.begin() + 12, ret.end()};
+		return util::h160(addr);
```

Comment:
Yeah... ideally ``util::h160`` would have a constructor for this purpose - and it has with ``bytesConstRef`` - but that should also be replaced anyways ;-)... so this is not ideal... but alright ;-)!

Reply:
Haha, only just saw https://github.com/ethereum/solidity/pull/10447#discussion_r536325882

Reply:
Those constructors only support left aligned truncation. Yeah, it is a giant mess. Not sure anymore what is the best approach cleaning this up, whatever we touch there's one more thing to make it nicer